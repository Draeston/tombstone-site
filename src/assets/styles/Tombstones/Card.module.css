/*noinspection CssUnknownTarget*/
@import url("https://fonts.googleapis.com/css2?family=IBM+Plex+Sans+Arabic:wght@200;300;400&display=swap");

.card {
  font-family: "IBM Plex Sans Arabic", sans-serif;
  padding: 10px;
  margin-right: 3%;
  margin-top: 40px;
  display: inline-block;
  text-align: center;

  width: 30%;
  background: white;
  box-shadow: black 5px 5px 20px;
  border-radius: 2px;
}

.img {
  height: 200px;
  width: auto;
}

.title {
  font-family: "IBM Plex Sans Arabic", sans-serif;
  font-size: 24px;
  font-weight: 400;
}

.price {
  font-family: "IBM Plex Sans Arabic", sans-serif;
  font-size: 32px;
  font-weight: 400;
}

.button {
  font-family: "IBM Plex Sans Arabic", sans-serif;
  border-radius: 20px;
  background: #005bff;
  padding: 10px 20px;
  color: white;
  border: none;
  margin-top: 10px;

  font-size: 18px;
  cursor: pointer;
}

.button:hover {
  background: #002d80;
}
