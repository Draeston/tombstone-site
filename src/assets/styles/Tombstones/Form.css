/*noinspection CssUnknownTarget*/
@import url("https://fonts.googleapis.com/css2?family=IBM+Plex+Sans+Arabic:wght@200;300;400&display=swap");

.order-form {
  text-align: center;
}

.form-title {
  font-size: 32px;
}

.form-name,
.form-phone,
.form-title,
.form-submit,
.form-item-name,
.form-img {
  display: block;
  margin: 0 auto 10px;
  font-family: "IBM Plex Sans Arabic", sans-serif;
}

input[type=text] {
  padding: 5px;
  outline: none;
  border-radius: 10px;
  border: 1px dotted gray;
  width: 90%;
  font-size: 16px;
}

.form-submit {
  border-radius: 20px;
  background: #005bff;
  padding: 10px 20px;
  color: white;
  border: none;
  font-size: 18px;
  text-transform: uppercase;
  letter-spacing: 3px;
  cursor: pointer;
}

.form-submit:hover {
  background: #0038a6;
}

.form-img {
  display: block;
  height: 200px;
}
