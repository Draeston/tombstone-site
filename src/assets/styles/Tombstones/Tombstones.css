.wrapper {
    padding: 0 calc(50px - 3%) 0 50px;
    background-image: linear-gradient( 109.6deg,  rgba(15,2,2,1) 11.2%, rgba(36,163,190,1) 91.1% );
}
