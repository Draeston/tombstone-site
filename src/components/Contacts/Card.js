import styles from "../../assets/styles/Contacts/Card.module.css";

export function Card(props) {
  return (
    <div className={styles.card}>
      <img className={styles.img} src={require(`${props.image}`)} />
      <div className={styles.textWrapper}>
        <h1>{props.name}</h1>
        <h2>@{props.telegram}</h2>
      </div>
    </div>
  );
}
