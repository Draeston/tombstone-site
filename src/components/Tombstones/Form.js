import "../../assets/styles/Tombstones/Form.css";
import { useState } from "react";
import InputMask from "react-input-mask";

export function Form(props) {
  const [name, setName] = useState("");
  const [phone, setPhone] = useState("");

  function handleSubmit(event) {
    const target = event.target;
    const named = target.name;
    const value = target.value;

    switch (named) {
      case "Name":
        setName(value);
        break;
      case "Phone":
        setPhone(value);
        break;
      default:
    }
  }

  function onSubmit(event) {
    event.preventDefault();
    if (phone.indexOf("_") !== -1 || phone === "") {
      return;
    }
    if (name === "") {
      return;
    }
    fetch("/api/order", {
      method: "POST",
      body: JSON.stringify({
        id: props.id,
        name,
        phone,
      }),
    }).then((res) => {
      if (res.ok) {
        alert("We will call you later");
      } else {
        alert("Error");
      }
    });
    props.closeModal();
  }

  return (
    <form onSubmit={onSubmit} className="order-form">
      <h1 className="form-title">Make an order</h1>
      <h2 className="form-item-name">Name: {props.title}</h2>
      <h2 className="form-item-name">Price: {props.price}</h2>
      <img
        className="form-img"
        src={require(`../../assets/images/tombstones/${props.img}`)}
        alt=""
      />
      <input
        className="form-name"
        placeholder="Your name"
        type="text"
        value={name}
        name="Name"
        onChange={(e) => handleSubmit(e)}
      />
      <InputMask
        className="form-phone"
        placeholder="Your phone"
        type="text"
        value={phone}
        name="Phone"
        onChange={(e) => handleSubmit(e)}
        maskChar="_"
        mask="+7 (999) 999-99-99"
      />
      <input className="form-submit" type="submit" value="Order" />
    </form>
  );
}
