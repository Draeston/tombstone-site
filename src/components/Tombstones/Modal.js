import { Form } from "./Form";
import * as ReactDOM from "react-dom";
import FocusTrap from "focus-trap-react";
import "../../assets/styles/Tombstones/Modal.css";

export function Modal(props) {
  return ReactDOM.createPortal(
    <FocusTrap>
      <aside
        role="dialog"
        tabIndex="-1"
        aria-modal="true"
        className="modal-cover"
        onClick={props.onClickOutside}
        onKeyDown={props.onKeyDown}
      >
        <div className="modal-area" ref={props.modalRef}>
          <button
            ref={props.buttonRef}
            aria-label="Close Modal"
            aria-labelledby="close-modal"
            className="_modal-close"
            type="button"
            onClick={props.closeModal}
          >
            <span id="close-modal" className="_hide-visual">
              Close
            </span>
            <svg className="_modal-close-icon" viewBox="0 0 40 40">
              <path d="M 10,10 L 30,30 M 30,10 L 10,30" />
            </svg>
          </button>
          <div className="modal-body">
            <Form
              img={props.img}
              title={props.title}
              price={props.price}
              id={props.id}
              closeModal={props.closeModal}
            />
          </div>
        </div>
      </aside>
    </FocusTrap>,
    document.body
  );
}
