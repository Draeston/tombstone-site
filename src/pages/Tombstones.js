import { Card } from "../components/Tombstones/Card";
import { useEffect, useState } from "react";
import "../assets/styles/Tombstones/Tombstones.css";

export function Tombstones() {
  const [list, setList] = useState([]);

  useEffect(() => {
    fetch("/api/tombstones")
      .then((response) => response.json())
      .then((data) =>
        setList(
          data.tombstones.map((item) => (
            <Card
              key={item.id}
              img={item.img}
              title={item.title}
              price={item.price}
              id={item.id}
            />
          ))
        )
      );
  });

  return <div className="wrapper">{list}</div>;
}
