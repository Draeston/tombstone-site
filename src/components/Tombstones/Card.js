import styles from "../../assets/styles/Tombstones/Card.module.css";
import { OrderButton } from "./OrderButton";
import { useState } from "react";
import { Modal } from "./Modal";

export function Card(props) {
  const [isShown, setIsShown] = useState(false);
  const [modal, setModal] = useState("");

  function showModal() {
    setIsShown(true);
  }

  function closeModal() {
    setIsShown(false);
  }

  function onKeyDown(event) {
    if (event.keyCode === 27) {
      closeModal();
    }
  }

  function onClickOutside(event) {
    if (modal && modal.contains(event.target)) {
      return;
    }
    closeModal();
  }

  return (
    <div className={styles.card}>
      <div>
        <img
          alt=""
          src={require(`../../assets/images/tombstones/${props.img}`)}
          className={styles.img}
        />
      </div>
      <p className={styles.title}>{props.title}</p>
      <p className={styles.price}>{props.price}</p>
      <OrderButton className={styles.button} showModal={showModal}>
        Make an order
      </OrderButton>
      {isShown ? (
        <Modal
          modalRef={(n) => setModal(n)}
          closeModal={closeModal}
          onKeyDown={onKeyDown}
          onClickOutside={onClickOutside}
          title={props.title}
          img={props.img}
          price={props.price}
          id={props.id}
        />
      ) : null}
    </div>
  );
}
