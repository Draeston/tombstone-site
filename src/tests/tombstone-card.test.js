import React from "react";
import renderer from "react-test-renderer"
import { Card } from "../components/Tombstones/Card";

test("testing tombstone card", () => {
  const tree = renderer
    .create(<Card img="1.png" title="Tombstone" price="800ла" id={1}/>)
    .toJSON()
  expect(tree).toMatchSnapshot();
})
