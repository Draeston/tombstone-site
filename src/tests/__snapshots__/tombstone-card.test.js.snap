// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`testing tombstone card 1`] = `
<div
  className="card"
>
  <div>
    <img
      alt=""
      className="img"
      src="1.png"
    />
  </div>
  <p
    className="title"
  >
    Tombstone
  </p>
  <p
    className="price"
  >
    800ла
  </p>
  <button
    className="button"
    onClick={[Function]}
    type="button"
  >
    Make an order
  </button>
</div>
`;
